modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/listeners/OpcUaSubscriptionLifecycleHandler.java

```diff
@@ -100,7 +101,7 @@ public OpcUaSubscriptionLifecycleHandler(
         return createNewSubscription(client)
                 .map(subscription -> {
                     subscription.setPublishingInterval((double) config.getOpcuaToMqttConfig().publishingInterval());
-                    subscription.setSubscriptionListener(new OpcUaSubscriptionLifecycleHandler(protocolAdapterMetricsService, tagStreamingService, eventService, adapterId, tags, client, dataPointFactory, config));
+                    subscription.setSubscriptionListener(this);
```

Comment:
This is the bug. The newly created handler is not registered so that it doesn't receive the keep-alive calls at all. Thus, every health check returns negative. The fix is to use `this` instead.